pkg/test/framework/components/echo/common/deployment/echos.go

```diff
@@ -75,8 +75,8 @@ func (c *Config) fillDefaults(ctx resource.Context) error {
 	}
 
 	// Verify the namespace for any custom deployments.
-	if c.Custom != nil {
-		for _, custom := range c.Custom.Get() {
+	if c.Configs != nil {
```

Comment:
Shouldn't this just be something like:

```golang
if c.Configs == nil {
  defaultConfigs := CreateDefaultEchoConfigs(*c, ctx)
  c.Configs = echo.ConfigFuture(&defaultConfigs)
}

for _, custom := range c.Configs.Get() {
  ...
}
```

Reply:
I don't this its needed -
1) Default echo configs won't be having any namespaces
2) We just need this check for real custom configs

The current implementation signifies that we verifying the namespaces for custom deployment, good for someone who is reading the code IMO

Reply:
I'm just proposing typical logic for handling defaults. Apply defaults first, then you don't have to ever consider it being nil from that point forward. It's actually simpler logic overall.

Reply:
I agree, done !!